### ISSUE TITLE
select throw errors：ERROR 1690 (22003): BIGINT UNSIGNED value is out of range in '(-254 + 0)'

### ISSUE THREAD

**Issue by lddlww** [pingcap/tidb#35972]
## Bug Report

select table throw errors：ERROR 1690 (22003): BIGINT UNSIGNED value is out of range in '(-254 + 0)'

### 1. Minimal reproduce step (Required)

no

### 2. What did you expect to see? (Required)
return data

### 3. What did you see instead (Required)
ERROR 1690 (22003): BIGINT UNSIGNED value is out of range in '(-254 + 0)'

### 4. What is your TiDB version? (Required)
5.3.0

**Comment by lddlww** [comment]
XOR return bigint unsigned result this problem

**Comment by aytrack** [comment]
mysql has the same  problem
```sql
mysql root@127.0.0.1:(none)> select sum(-254+1^1);
(1690, "BIGINT UNSIGNED value is out of range in '(-(254) + (1 ^ 1))'")
mysql root@127.0.0.1:(none)> select version();
+-----------+
| version() |
+-----------+
| 8.0.29    |
+-----------+
```

**Comment by zanmato1984** [comment]
> mysql has the same problem
> 
> ```sql
> mysql root@127.0.0.1:(none)> select sum(-254+1^1);
> (1690, "BIGINT UNSIGNED value is out of range in '(-(254) + (1 ^ 1))'")
> mysql root@127.0.0.1:(none)> select version();
> +-----------+
> | version() |
> +-----------+
> | 8.0.29    |
> +-----------+
> ```

So this is not a bug?

**Comment by zanmato1984** [comment]
> ![image](https://user-images.githubusercontent.com/27006237/177523391-bfb62446-08f3-4241-9645-127b810064c2.png) seems `tiflash`(`clickhouse`) return ok

If you directly ran TiFlash client, it goes to the original ClickHouse logic, which considers this as a valid case. But TiFlash is part of TiDB and supposed to run tightly within TiDB, i.e., **NOT** using TiFlash client. And TiDB respects MySQL logic, which considers this as an error. So this is expected and not a bug.

**Comment by zanmato1984** [comment]
> > And TiDB respects MySQL logic, which considers this as an error
> 
> so this is database level or software behavior？not all database throw the problem

Right. SQL is not a fully standardized language. In fact it is highly implementation dependent. You can easily find tons of cases that differ from one database to another.

**Comment by lddlww** [comment]
if this problem can be fixed？ i think tidb will not 100% compatible with mysql

**Comment by zanmato1984** [comment]
> if this problem can be fixed？

I don't think we will fix this as we don't think this is a bug considering we have the same behavior as MySQL. Maybe someday MySQL thinks this is a bug then we will fix it.

> i think tidb will not 100% compatible with mysql

Yeah it's so true that we are not 100% MySQL compatible, but that doesn't give the reason to modify something that is already compatible with MySQL. Again, we only consider a behavior **different** than MySQL a (potential) bug.

If you are really concerned with this case, you can try the following statement:
```
select sum(-254+cast(1^1 as signed));
```

**Comment by lddlww** [comment]
> Yeah it's so true that we are not 100% MySQL compatible, but that doesn't give the reason to modify something that is already compatible with MySQL. Again, we only consider a behavior **different** than MySQL a (potential) bug

thanks for reply ，and i am closing this issue now，please feel free to open this issue if you want
